### ISSUE TITLE
How to set up with Azure / O365

### ISSUE THREAD

**Issue by variamus** [edgardmessias/glpi-singlesignon#39]
Hello,
I've tried to find guidelines about how to set up but I'm still unable to do so.
I've followed this procedure: Azure - https://docs.microsoft.com/azure/app-service/configure-authentication-provider-aad
I couldn't go for option 1 as I don't have the menu to Add identity provider showing up  and while going for option 2, I get the error that the provider link isn't correct.
Does anyone have configured already and would be kind enough to guide me through ?

Thanks !

**Comment by raphou91300** [comment]
Hi, 

I have the same issue, 

Someone ?

**Comment by kbron94** [comment]
Hello, 

for that u have to log in to Azure AD. 
After that u click on the left side on "App registrations". Here u can search for your App or create it. 
And here u can click on the page "Authentication". 

BR

**Comment by eduardomozart** [comment]
You can follow the steps from https://docs.axissecurity.com/docs/azure-idp-integration to configure your Azure AD integration.
Just make sure to use the "Callback URL" available from plug-in provider settings into your Azure AD application, as seen below:
<img width="826" alt="Image" src="https://github.com/user-attachments/assets/8ea51d3c-f650-4791-a1c5-4aca17bb2158" />

Here's my working Microsoft 365 Admin Center config:

<img width="1441" alt="Image" src="https://github.com/user-attachments/assets/9c463c4e-301a-44c8-a9b9-d9df4e763add" />

**Comment by Daeloran** [comment]
Hi @eduardomozart , I did the same configurations but I get the error "User not authorize to connect to GLPI".
![Image](https://github.com/user-attachments/assets/e7fa8836-6996-4482-90ef-874cb8898b3c)

Did I miss one step ? 
It's annoying not to have documentation even if only to indicate the confs and elements expected on the plugin side

Must I create a user before ? it's a loss if it's needed to do that as prerequisite :(

Thank you for your reply !

**Comment by Daeloran** [comment]
Ok so for information. I don't know why, but I can't use Single tenant in my Azure configuration 
![Image](https://github.com/user-attachments/assets/c604e802-7642-4522-b503-e63c7a7bad1a)
Just like this, I get my previous error, but when I add my domain to the plugins configuration as "AuthorizedDomains" it work.

If possible I prefere to use Single Tenant. Do you have some clues ? I get an error telling me  I need to provide a "endpoint" to use single tenant but don't really know/understand where to define this.

Thank you for your reply :)

**Comment by Daeloran** [comment]
Hi,

Little update, I was able to fix my previous issue using "Generic" instead of "Azure" type of SSO type.
Don't understand why.
I also noticed that I still have the problem if I leave the dopmain/GLPI domain mapping to "no" (the default).
I also notice that even when using the option to use the email address as an identifier, it is not the @ that is used

Thank you for your help

**Comment by Daeloran** [comment]
I comeback again after few more tests.
I tried Generic mode without the parameter I shown on my previous screen an it works and get the email address as wanted + the email is now used as username.

So I think my issue is solved.
Do you mind if I detail my case in the wiki ? I'm happy to be able to use a free project which work and is active so I want to pay back as I can :)

**Comment by eduardomozart** [comment]
> Hello [@eduardomozart](https://github.com/eduardomozart), Here configuration i talked about (sorry it's in french)
> 
> ![Image](https://github.com/user-attachments/assets/4e97a1b4-436e-4858-8b26-c239ae67d1d8)
> 
> For my question about "Use Email as Login" I've misunderstand the suage of this. I though it was for use the mail address as username when the plugins create a new user from Azure.
> 
> I've other question but I'll open new issues yes thank you :)

The use of the email address should be the default behavior when auto-creating the user if it doesn't exist. It only will not create the user using the mail address as username if an user with the same mail already exists on GLPI user database.

Of course, feel free to open a ticket for any issue you found on this plugin.

**Comment by eduardomozart** [comment]
> I comeback again after few more tests. I tried Generic mode without the parameter I shown on my previous screen an it works and get the email address as wanted + the email is now used as username.
> 
> So I think my issue is solved. Do you mind if I detail my case in the wiki ? I'm happy to be able to use a free project which work and is active so I want to pay back as I can :)

Of course. (1) You can write it here on comments and I create a wiki page with your steps or (2) You can fork this repository, edit the wiki locally at https://github.com/Daeloran/glpi-singlesignon/wiki and create a PR at https://github.com/edgardmessias/glpi-singlesignon/pulls and we integrate your changes directly.
